# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositorD3D11SWGL.h
# Commit: 81601a02d9b7
# Full Hash: 81601a02d9b7e658f37f9525cce902eed94d678f
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2020-11-05 16:09:03
# Regressor Bug: 1673983
# File Overlap Count: 2
# Description:
#   Bug 1673983 - Support using D3D11 textures as external compositor surfaces with D3D11 compositor. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95800
# ==============================================================================

diff -r 72d4bf5b093e -r 81601a02d9b7 gfx/webrender_bindings/RenderCompositorD3D11SWGL.h
--- a/gfx/webrender_bindings/RenderCompositorD3D11SWGL.h	Thu Nov 05 09:15:05 2020 +0000
+++ b/gfx/webrender_bindings/RenderCompositorD3D11SWGL.h	Thu Nov 05 09:15:28 2020 +0000
@@ -68,13 +68,12 @@
   void UnmapTile() override;
   void CreateSurface(wr::NativeSurfaceId aId, wr::DeviceIntPoint aVirtualOffset,
                      wr::DeviceIntSize aTileSize, bool aIsOpaque) override;
-  void CreateExternalSurface(wr::NativeSurfaceId aId, bool aIsOpaque) override {
-  }
+  void CreateExternalSurface(wr::NativeSurfaceId aId, bool aIsOpaque) override;
   void DestroySurface(NativeSurfaceId aId) override;
   void CreateTile(wr::NativeSurfaceId, int32_t aX, int32_t aY) override;
   void DestroyTile(wr::NativeSurfaceId, int32_t aX, int32_t aY) override;
   void AttachExternalImage(wr::NativeSurfaceId aId,
-                           wr::ExternalImageId aExternalImage) override {}
+                           wr::ExternalImageId aExternalImage) override;
   void AddSurface(wr::NativeSurfaceId aId,
                   const wr::CompositorSurfaceTransform& aTransform,
                   wr::DeviceIntRect aClipRect,
@@ -133,6 +132,7 @@
     bool mIsOpaque;
     bool mIsExternal = false;
     std::unordered_map<TileKey, Tile, Tile::KeyHashFn> mTiles;
+    RefPtr<RenderTextureHost> mExternalImage;
 
     struct IdHashFn {
       std::size_t operator()(const wr::NativeSurfaceId& aId) const {